sys/compat/freebsd64/freebsd64_vfs.c

```diff
@@ -916,21 +916,18 @@ freebsd64_link(struct thread *td, struct freebsd64_link_args *uap)
 {
 
 	return (kern_linkat(td, AT_FDCWD, AT_FDCWD, __USER_CAP_STR(uap->path),
-	    __USER_CAP_STR(uap->to), UIO_USERSPACE, FOLLOW));
+	    __USER_CAP_STR(uap->to), UIO_USERSPACE, AT_SYMLINK_FOLLOW));
 }
 
 int
 freebsd64_linkat(struct thread *td, struct freebsd64_linkat_args *uap)
 {
-	int flag;
 
-	flag = uap->flag;
-	if (flag & ~AT_SYMLINK_FOLLOW)
+	if ((uap->flag & ~(AT_SYMLINK_FOLLOW | AT_RESOLVE_BENEATH)) != 0)
```

Comment:
Can/should all these be pushed into the kern_* functions now?

Reply:
That's a good question.  I had thought about it as right now it's kind of a split in that some kern_foo functions check for invalid flags and some do it in sys_foo instead.  I'll look at this again to see if there's consensus to do it in kern_foo.

Reply:
Done.